src/atomate2/common/schemas/defects.py

```diff
@@ -1,16 +1,112 @@
 import logging
-from typing import Any, Callable, List, Optional, Tuple, Type
+from typing import Any, Callable, Dict, List, Optional, Tuple, Union
 
 import numpy as np
 from pydantic import BaseModel, Field
+from pymatgen.analysis.defects.core import Defect
+from pymatgen.analysis.defects.thermo import DefectEntry, FormationEnergyDiagram
 from pymatgen.core import Structure
-from pymatgen.entries.computed_entries import ComputedStructureEntry
+from pymatgen.entries.computed_entries import ComputedEntry, ComputedStructureEntry
 
 from atomate2.vasp.schemas.task import TaskDocument
 
 logger = logging.getLogger(__name__)
 
-__all__ = ["CCDDocument"]
+
+class FormationEnergyDiagramDocument(BaseModel):
+    """A document for storing a formation energy diagram.
+    Basically a PyDantic version of the `FormationEnergyDiagram` dataclass
+    with some additional data fields.
+    The `pd_entries` field is now optional since the workflow will not necessarily
+    have all the entries in the phase diagram computed.
+    """
+
+    bulk_entry: ComputedStructureEntry = Field(
+        None, description="The ComputedEntry representing the bulk structure."
+    )
+
+    defect_entries: List[DefectEntry] = Field(
+        None, description="The defect entries for the formation energy diagram."
+    )
+
+    pd_entries: List[ComputedEntry] = Field(
+        None, description="The entries used to construct the phase diagram."
+    )
+
+    vbm: float = Field(
+        None, description="The VBM of the pristine supercell calculation."
+    )
+
+    band_gap: float = Field(
+        None, description="The band gap of the pristine supercell calculation."
+    )
+
+    inc_inf_values: bool = Field(
+        None, description="Whether or not to include infinite values in the diagram."
+    )
+
+    defect: Defect = Field(
+        None, description="The defect for which the diagram is being calculated."
+    )
+
+    bulk_sc_dir: str = Field(
+        None, description="The directory name of the pristine supercell calculation."
+    )
+
+    defect_sc_dirs: Dict[int, str] = Field(
+        None, description="The directory names of the charged defect calculations."
+    )
+
+    dielectric: Union[float, List[List[float]]] = Field(
+        None,
+        description="The dielectric constant or tensor, can be used to compute "
+        "finite-size corrections.",
+    )
+
+    @classmethod
+    def from_FormationEnergyDiagram(
```

Comment:
To follow python style can you change this to:

```suggestion
    def from_formation_energy_diagram(
```